onedal/linear_model/linear_model.py

```diff
@@ -198,8 +161,43 @@ def __init__(
         super().__init__(fit_intercept=fit_intercept, copy_X=copy_X, algorithm=algorithm)
 
     def fit(self, X, y, queue=None):
-        return super()._fit(X, y, _backend.linear_model.regression, queue)
+        module = _backend.linear_model.regression
 
-    def predict(self, X, queue=None):
-        y = super()._predict(X, _backend.linear_model.regression, queue)
-        return y
+        policy = self._get_policy(queue, X, y)
+
+        X_loc, y_loc = X, y
+        if not isinstance(X, np.ndarray):
+            X_loc = np.asarray(X)
```

Comment:
X_loc looks to be unnecessary.
```suggestion
        X = np.asanarray(X)
```
This will guarantee that it should have a dtype attribute.

Reply:
it is not my code. I just rearranged lines Nikita used. I agree that it is unnecessary but since this PR is not related to LinearRegression class I tried to do minimal changes to this class. Probably it makes sense to refactor LinearRegression later

Reply:
I think @samir-nasibli is correct then in this case in https://github.com/intel/scikit-learn-intelex/pull/1656#pullrequestreview-1837303634 . By moving things around you are refactoring the code, you can do it separately in a different PR, or you take the full responsibility in this one for both tasks.

Reply:
i don't really think it is the same. in my current changes i kept all variables and external methods calls and their call order the same. i just removed from BaseLinearRegressionClass methods which are not necessary in IncrementalLinearRegression and put it to LinearRegression class. more serious code changes probably really have to be perf-measured.